sys/init.c

```diff
@@ -577,28 +577,49 @@ DokanGetMountPointList(__in PREQUEST_CONTEXT RequestContext) {
   PDOKAN_MOUNT_POINT_INFO dokanMountPointInfo;
   USHORT i = 0;
 
-  __try {
-    ExAcquireResourceExclusiveLite(
-        &RequestContext->DokanGlobal->MountPointListLock, TRUE);
-    dokanMountPointInfo = (PDOKAN_MOUNT_POINT_INFO)
-                              RequestContext->Irp->AssociatedIrp.SystemBuffer;
-    for (listEntry = RequestContext->DokanGlobal->MountPointList.Flink;
-         listEntry != &RequestContext->DokanGlobal->MountPointList;
-         listEntry = listEntry->Flink, ++i) {
+  BOOLEAN isAdmin =
+      SeSinglePrivilegeCheck(RtlConvertLongToLuid(SE_IMPERSONATE_PRIVILEGE),
+                             RequestContext->Irp->RequestorMode);
+
+  ULONG sessionId;
+  if (!NT_SUCCESS(IoGetRequestorSessionId(RequestContext->Irp, &sessionId))) {
+    sessionId = 0;
+  }
+
+  status = STATUS_SUCCESS;
+
+  ExAcquireResourceExclusiveLite(
+    &RequestContext->DokanGlobal->MountPointListLock, TRUE);
+
+  dokanMountPointInfo =
+    (PDOKAN_MOUNT_POINT_INFO)RequestContext->Irp->AssociatedIrp.SystemBuffer;
+  
+  for (listEntry = RequestContext->DokanGlobal->MountPointList.Flink;
+       listEntry != &RequestContext->DokanGlobal->MountPointList;
+       listEntry = listEntry->Flink, ++i) {
+
+    mountEntry = CONTAINING_RECORD(listEntry, MOUNT_ENTRY, ListEntry);
+    ExAcquireResourceSharedLite(&mountEntry->Resource, TRUE);
+
+    __try {
+      // For non-admins, skip drives that are not in same session as
+      // requestor
```

Comment:
nit: This does not fit on one line ?

Reply:
Good point. Old habits to always line break at 79 characters, but that is pretty much pointless these days. I'll change it.